junit-platform-engine/src/main/java/org/junit/platform/engine/discovery/NestedClassSelector.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2015-2019 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.engine.discovery;
+
+import static java.util.stream.Collectors.toList;
+import static org.apiguardian.api.API.Status.STABLE;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.apiguardian.api.API;
+import org.junit.platform.commons.PreconditionViolationException;
+import org.junit.platform.engine.DiscoverySelector;
+
+/**
+ * A {@link DiscoverySelector} that selects a nested {@link Class}
+ * or class name enclosed in other classes so that
+ * {@link org.junit.platform.engine.TestEngine TestEngines} can discover
+ * tests or containers based on classes.
+ *
+ * <p>If Java {@link Class} references are provided for the nested class or
+ * the enclosing classes, the selector will return these {@code Class} and
+ * their class names accordingly. If class names are provided, the selector
+ * will only attempt to lazily load the {@link Class} if
+ * {@link #getEnclosingClasses()} or {@link #getNestedClass()} are invoked.
+ *
+ * <p>In this context, Java {@link Class} means anything that can be referenced
+ * as a {@link Class} on the JVM &mdash; for example, classes from other JVM
+ * languages such Groovy, Scala, etc.
+ *
+ * @since 1.6
+ * @see DiscoverySelectors#selectNestedClass(List, Class)
+ * @see DiscoverySelectors#selectNestedClass(List, String)
+ * @see org.junit.platform.engine.support.descriptor.ClassSource
```

Comment:
Should we point to `ClassSelector` as well?

Reply:
You're right! 👍